utils/post_processing/backconvert_borealis.py

```diff
@@ -0,0 +1,204 @@
+		# format dictionary key in the same way it is done
+		# in datawrite on site
+		sqn_dt_ts = datetime.datetime.utcfromtimestamp(seq_ts[0])
+		epoch = datetime.datetime.utcfromtimestamp(0)
+		key = str(int((sqn_dt_ts - epoch).total_seconds() * 1000))
+		
+		ts_dict[key] = dict()
+		for f in data_record:
+			if not type(data_record[f]) is np.ndarray:
+				ts_dict[key][f] = data_record[f]
+			else:
+				if np.shape(data_record[f])[0] == num_records:
+					# pass data fields that are written per record
+					pass
+				else:
+					ts_dict[key][f] = data_record[f]
+		# Handle per record fields
+		num_sequences = data_record["num_sequences"][rec]
+		ts_dict[key]["num_sequences"] = num_sequences
+		ts_dict[key]["int_time"] = data_record["int_time"][rec]
+		ts_dict[key]["sqn_timestamps"] = data_record["sqn_timestamps"][rec, 0:int(num_sequences)]
+		ts_dict[key]["noise_at_freq"] = data_record["noise_at_freq"][rec, 0:int(num_sequences)]
+		ts_dict[key]["data_descriptors"] = ts_dict[key]["data_descriptors"][1:]
+		ts_dict[key]["data_dimensions"] = data_record["data_dimensions"][rec]
+
+		ts_dict[key]["data"] = np.trim_zeros(data_record["data"][rec].flatten())
+	
+	return ts_dict
+
+def backconvert_rawacf(data_record):
+	"""
+	Converts a restructured raw acf file back to its
+	original site format
+	Args:
+		data_record (dict): An opened rawacf hdf5 file
+	"""
+	num_records = len(data_record["int_time"])
+	ts_dict = dict()
+	# get keys from first sequence timestamps
+	for rec, seq_ts in enumerate(data_record["sqn_timestamps"]):
+		# format dictionary key in the same way it is done
+		# in datawrite on site
+		sqn_dt_ts = datetime.datetime.utcfromtimestamp(seq_ts[0])
+		epoch = datetime.datetime.utcfromtimestamp(0)
+		key = str(int((sqn_dt_ts - epoch).total_seconds() * 1000))
+		
+		ts_dict[key] = dict()
+		for f in data_record:
+			if not type(data_record[f]) is np.ndarray:
+				ts_dict[key][f] = data_record[f]
+			else:
+				if np.shape(data_record[f])[0] == num_records:
+					# pass data fields that are written per record
+					pass
+				else:
+					ts_dict[key][f] = data_record[f]
+		# Handle per record fields
+		num_sequences = data_record["num_sequences"][rec]
+		ts_dict[key]["num_sequences"] = num_sequences
+		ts_dict[key]["int_time"] = data_record["int_time"][rec]
+		ts_dict[key]["sqn_timestamps"] = data_record["sqn_timestamps"][rec, 0:int(num_sequences)]
+		ts_dict[key]["noise_at_freq"] = data_record["noise_at_freq"][rec, 0:int(num_sequences)]
+		ts_dict[key]["correlation_descriptors"] = ts_dict[key]["correlation_descriptors"][1:]
+
+		ts_dict[key]["main_acfs"] = data_record["main_acfs"][rec].flatten()
+		ts_dict[key]["intf_acfs"] = data_record["intf_acfs"][rec].flatten()
+		ts_dict[key]["xcfs"] = data_record["xcfs"][rec].flatten()
+		
+	return ts_dict
+
+
+def write_backconverted(ts_dict, data_path):
+	"""
+	Writes a set of back-converted borealis data to file in the original site
+	file format
+	"""
+	temp_file = 'temp.hdf5'
```

Comment:
Dangerous in some cases having a hard-coded temp file name, I suggest the tempfile library: https://stackoverflow.com/questions/10501247/best-way-to-generate-random-file-names-in-python OR the uuid library for generating a random string for the filename.